src/generic_ui/polymer/tip.ts

```diff
@@ -0,0 +1,64 @@
+Polymer({
+  close: function() {
+    this.fire('closed');
+    this.remove();
+  },
+  doReposition: function() {
+    var prev = this.previousSibling;
+    if (prev.offsetLeft === 0 && prev.offsetTop === 0 && prev.offsetWidth === 0
+        && prev.offsetHeight === 0) {
+      /* if the DOM is not set up yet, keep checking every half second */
```

Comment:
The comment says "if the DOM is not set up yet" - but doesn't the domReady callback tell you when the DOM is setup?

Also right now we are always creating all of our Polymer elements, and then toggling the hidden property based on whether they should be shown.  Maybe this is a case where it would be better to to dynamically add/remove elements from the dom instead of using hidden?